src/docs/self-hosted/sso.mdx

```diff
@@ -0,0 +1,117 @@
+- Via a third-party service which implements an authentication pipeline
+
+
+## Using a middleware proxy (SAML2)
+
+As of Sentry 20.6.0, self-hosted Sentry comes with built-in support for SAML2 and certain auth providers. For older versions you will need to add the following line into [`sentry/requirements.txt`](https://github.com/getsentry/onpremise/blob/10.0.1/sentry/requirements.example.txt) before running `./install.sh`:
+
+```
+sentry-auth-saml2@https://github.com/getsentry/sentry-auth-saml2/archive/master.zip#egg=sentry-auth-saml2
+```
+
+ The way you can set this up is the same as [sentry.io](https://sentry.io) except, you need to use your own instance's [`url-prefix`](https://develop.sentry.dev/config/#general) for the URLs mentioned in the documentation.
+
+Please refer to our [main SAML documentation](https://docs.sentry.io/accounts/sso/#saml2-identity-provider) for all the details.
+
+## SSO with OAuth
+
+<Alert title="Note" level="info">
+    Once you enable SSO, this will be the only way to login to your self-hosted instance. If you need free registration along with SSO, you can comment on the <Link to="https://github.com/getsentry/sentry/pull/16247">GitHub PR</Link> for this.
+</Alert>
+
+### Google Auth
+
+As of [Sentry 9.1](https://github.com/getsentry/onpremise/releases/tag/9.1.2), self-hosted Sentry comes with built-in Google Auth support. To enable, you'll need to [create a client ID and secret for your Google App](https://developers.google.com/identity/sign-in/web/server-side-flow#step_1_create_a_client_id_and_client_secret) and then enter these values into your [`sentry/config.yaml`](https://github.com/getsentry/onpremise/blob/master/sentry/config.example.yml) file respectively:
+
+```yaml
+auth-google.client-id: '<client id>'
+auth-google.client-secret: '<client secret>'
+```
+
+<Alert title="Note" level="info">
+    Remember that you'd need to restart all Sentry services once you change your settings. See the <Link to="/self-hosted/#configuration">configuration section</Link> for more information.
+</Alert>
+
+### GitHub Auth
+
+As of [Sentry 10](https://github.com/getsentry/onpremise/releases/tag/10.0.1), self-hosted Sentry comes with built-in GitHub Auth support. To enable, you'll need to create a <GitHubOrgLink url="https://github.com/organizations/:org/settings/apps/new?name=:org-SentryIntegration&public=false&members=read&emails=read&administration=read&contents=read&issues=write&pull_requests=write&repository_hooks=write&url=https%3A%2F%2Fyour-sentry-url-prefix.com&callback_url=https%3A%2F%2Fyour-sentry-url-prefix.com%2Fauth%2Fsso%2F&setup_url=https%3A%2F%2Fyour-sentry-url-prefix.com%2Fextensions%2Fgithub%2Fsetup%2F&webhook_url=https%3A%2F%2Fyour-sentry-url-prefix.com%2Fextensions%2Fgithub%2Fwebhook%2F&events[]=push&events[]=pull_request&webhook_active=true" defaultOrg="your-organization">new GitHub App under </GitHubOrgLink> and install it.
```

Comment:
Open to all kinds of feedback regarding how to improve the UX here. //cc @billyvg

Reply:
Yeah I agree with @untitaker - it would help if it was more "form"-like as currently I didn't even know that was an input box.

Reply:
That's pretty cool. @dcramer do we have some form styles in docs we can use here? Do you know? Maybe @cameronmcefee

Reply:
Ideally we'll have this inside Sentry itself so we don't need to ask for `url-prefix` manually. One step at a time though :)

Reply:
To my knowledge there are no dedicated form styles, since it was never anticipated. We do have an input style on the search box and the primary CTA button style which could be copied for this purpose.

This section definitely needs some work to read like a form, but I unfortunately cannot offer support at this time.